backend/Services/DataBuilder.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Services;
+
+use Internships\Interfaces\BuildTool;
+use Internships\Interfaces\SerializableInfo;
+use Internships\Models\PathPair;
+
+abstract class DataBuilder extends DataSanitizer implements BuildTool, SerializableInfo
```

Comment:
Are you sure that `DataSanitizer` should be the parent of `DataBuilder`?

The name suggests different responsibilities of these classes.
I would use `DataSanitizer` as a dependency of `DataBuilder` using composition rather than inheritance.